### ISSUE TITLE
pip-compile cannot compile elyra-python-editor-extension

### ISSUE THREAD

**Issue by samuelvl** [elyra-ai/elyra#2974]
**Describe the issue**
Dependencies for elyra-python-editor-extension package cannot be generated with `pip-compile`.

**To Reproduce**

Create a `requirements.in` with the following content:

```shell
elyra-python-editor-extension==3.12.0
```

Run the `pip-compile` tool:

```shell
$ pip-compile --generate-hashes requirements.in 
Could not find a version that matches pycodestyle<2.9.0,>=2.8.0,>=2.9.1 (from autopep8==1.7.0->elyra-server==3.12.0->elyra-python-editor-extension==3.12.0->-r requirements.in (line 1))
Tried: 2.0.0, 2.0.0, 2.1.0, 2.1.0, 2.2.0, 2.2.0, 2.3.0, 2.3.0, 2.3.1, 2.3.1, 2.4.0, 2.4.0, 2.5.0, 2.5.0, 2.6.0, 2.6.0, 2.7.0, 2.7.0, 2.8.0, 2.8.0, 2.9.0, 2.9.0, 2.9.1, 2.9.1
Skipped pre-versions: 1.8.0.dev0, 1.8.0.dev0, 2.0.0a1, 2.0.0a1, 2.6.0a1, 2.6.0a1
There are incompatible versions in the resolved dependencies:
  pycodestyle>=2.9.1 (from autopep8==1.7.0->elyra-server==3.12.0->elyra-python-editor-extension==3.12.0->-r requirements.in (line 1))
  pycodestyle<2.9.0,>=2.8.0 (from python-lsp-server[all]==1.5.0->elyra-server==3.12.0->elyra-python-editor-extension==3.12.0->-r requirements.in (line 1))
```

**Expected behavior**
Do not return an error.

**Deployment information**
Describe what you've deployed and how:
- **Python version 3.8.13**
- **pip-compile version 6.8.0**

**Comment by akchinSTC** [comment]
@samuelvl - we are transitioning to using the pyproject.toml and updating dependencies as needed. I noted several issues when generating the requirements lockfile with pip-compile using this new format including this issue, but they seem to have been resolved. See https://github.com/elyra-ai/elyra/pull/3041

**Comment by lresende** [comment]
Duplicate or will be resolved by #3041
